src/bilstm_train_regression.py

```diff
@@ -0,0 +1,256 @@
+
+
+class RocAucEvaluation(Callback):
+    def __init__(self, validation_data=(), interval=1):
+        super(Callback, self).__init__()
+
+        self.interval = interval
+        self.X_val, self.y_val = validation_data
+
+    def on_epoch_end(self, epoch, logs={}):
+        if epoch % self.interval == 0:
+            y_pred = self.model.predict(self.X_val, verbose=0)
+            score = roc_auc_score(self.y_val, y_pred)
+            print("\n ROC-AUC - epoch: {:d} - score: {:.6f}".format(epoch+1, score))
+
+
+if __name__ == '__main__':
+
+    args = get_args()
+    logger.info(f"Reading test file")
+    test = process_data_regression(args.input_path, args.test_filename)
+    X_test = test.drop(args.factor, axis=1)
+    texts_test = X_test["Text-EN"].str.lower()
+    cards_test = X_test.filter(regex="card_.*")  # Using a regular expression cause some cards may not be in test.
+    y_test = test[args.factor]
+    logger.info(f" Using test dataset with {X_test.shape[0]} instances")
+
+    logger.info(f"Reading train file")
+    train = process_data_regression(args.input_path, args.train_filename)
+    X_train = train.drop(args.factor, axis=1)
+    texts_train = X_train["Text-EN"].str.lower()
+    cards_train = X_train.filter(regex="card_.*")  # Using a regular expression cause some cards may not be in train.
+    y_train = train[args.factor]
+    y_train_norm = (y_train - y_train.mean())/y_train.std()
+    y_test_norm = (y_test - y_train.mean())/y_train.std()
+    logger.info(f" Using train dataset with {X_train.shape[0]} instances")
+
+    tok = get_tokenizer(args)
+    tok.fit_on_texts(list(texts_train) + list(texts_test))
+    texts_train = tok.texts_to_sequences(texts_train)
+    texts_test = tok.texts_to_sequences(texts_test)
+    x_train = sequence.pad_sequences(texts_train, maxlen=args.max_length)
+    x_test = sequence.pad_sequences(texts_test, maxlen=args.max_length)
+
+    embeddings_index = {}
+    with open(os.path.join(args.embedding_path, args.embedding_file), encoding='utf8') as f:
+        for line in f:
+            values = line.rstrip().rsplit(' ')
+            word = values[0]
+            coefficients = np.asarray(values[1:], dtype='float32')
+            embeddings_index[word] = coefficients
+
+    word_index = tok.word_index
+    # prepare embedding matrix
+    num_words = min(args.max_features, len(word_index) + 1)
+    embedding_matrix = np.zeros((num_words, args.embed_size))
+    for word, i in word_index.items():
+        if i >= args.max_features:
+            continue
+        embedding_vector = embeddings_index.get(word)
+        if embedding_vector is not None:
+            # words not found in embedding index will be all-zeros.
+            embedding_matrix[i] = embedding_vector
+
+    text_input = Input(shape=(args.max_length,), name='text_input')
+    x = Embedding(args.max_features, args.embed_size, weights=[embedding_matrix], trainable=False)(text_input)
+    x = SpatialDropout1D(0.2)(x)
```

Comment:
remove this

Reply:
What do you mean? That's the model we are using

Reply:
The Line: ` x = SpatialDropout1D(0.2)(x)` That dropout doesn't make a lot of sense like in the other model